overview.html

```diff
@@ -0,0 +1,452 @@
+<html>
+<!--
+This overview.html provides Javadoc overview for the application.
+The content from the body section is added to the Javadoc overview page.
+There is no need to add a head/title as that is provided by Javadoc.
+-->
+<body>
+
+<P>
+    This is the official Cloudant library for java.
+</P>
+
+<OL>
+    <LI><a href="#Initialization">Initialization</a>
+        <OL>
+            <LI><a href="#Cloudant service example">Cloudant service example</a></LI>
+            <LI><a href="#Cloudant Local example">Cloudant Local example</a></LI>
+        </OL>
+    </LI>
+    <LI><a href="#Authentication">Authentication</a>
+        <OL>
+            <LI><a href="#Cookie authentication">Cookie authentication</a></LI>
+            <LI><a href="#Custom authentication">Custom authentication</a></LI>
+        </OL>
+    </LI>
+    <LI><a href="#Functions">Functions</a>
```

Comment:
This was copy/paste from the previous API reference. Not all of them are methods, `Cloudant Query`, for example. I agree that perhaps function is a misleading word in this context. What about some mix of capabilities and operations e.g.

* Capabilities
    * Server Operations
    * Database Operations
    * Document Operations
        * Single document CRUD
        * Bulk document CRU
    * Attachments
        * Standalone Attachments
        * Inline attachment
    * Design Documents
    * Using Views
    * Cloudant Query
    * Cloudant Search